File: app/src/main/java/ru/myitschool/work/SAPI.java
package ru.myitschool.work;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.Part;
import retrofit2.http.Path;

public interface SAPI {
    @GET("api/{login}/auth")
    Call<Void> checkLogin(@Path("login") String login);

    @GET("/api/{login}/info")
    Call<User> getInfo(@Path("login") String login);

    @PATCH("/api/{login}/open")
    Call<Void> openDoor(@Path("login") String login, @Body RequestBodyModel body);
}

File: app/src/main/java/ru/myitschool/work/data/SAPI.kt
package ru.myitschool.work.data

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.Path
import ru.myitschool.work.RequestBodyModel
import ru.myitschool.work.data.dto.UserDTO

interface SAPI {
    @GET("/api/{login}/info")
    fun getInfo(@Path("login") login: String): Call<UserDTO>

    @PATCH("/api/{login}/open")
    fun openDoor(@Path("login") login: String, @Body body: RequestBodyModel): Call<Void>
}